tests/dashbio_demos/dash-ngl/app.py

```diff
@@ -0,0 +1,238 @@
+import os,glob
+
+import dash
+from dash.dependencies import Input, Output
+import dash_html_components as html
+import dash_core_components as dcc
+import dash_daq as daq
+import dashbio
+
+try:
+    from layout_helper import run_standalone_app
+except ModuleNotFoundError:
+    from .layout_helper import run_standalone_app
+
+color_list = [
+    "#e41a1c",
+    "#377eb8",
+    "#4daf4a",
+    "#984ea3",
+    "#ff7f00",
+    "#ffff33",
+    "#a65628",
+    "#f781bf",
+    "#999999",
+]
+
+data_dict = {
+    "selectedValue": "placeholder",
+    "chain": "ALL",
+    "color": "#e41a1c",
+    "filename": "placeholder",
+    "ext": "",
+    "config": {"type": "", "input": ""},
+}
+
+dropdown_options = [
+    "1PNK",
+    "5L73",
+    "4GWM",
+    "3L9P",
+    "5L73.A_4GWM.A_3L9P.A",
+    "6CHG",
+    "6CHG.A",
+    "3K8P",
+    "6CHG.A_3K8P.D",
+    "2MRU",
+    "1BNA",
+    "2MRU_1BNA",
+]
+
+# custom functions
+def getData(selection, pdb_id, color):
+
+    chain = "ALL"
+
+    # check if only one chain should be displayed
+    if "." in pdb_id:
+        pdb_id, chain = pdb_id.split(".")
+
+    # get path to protein structure
+    fname = [f for f in glob.glob("data/" + pdb_id + ".*")][0]
+
+    ext = fname.split(".")[-1]
+    with open(fname, "r") as f:
+        contents = f.read()
+
+    return {
+        "selectedValue": selection,
+        "chain": chain,
+        "color": color,
+        "filename": fname.split("/")[-1],
+        "ext": ext,
+        "config": {"type": "text/plain", "input": contents},
+    }
+
+
+###Define app layout
+label_width = 4
+col_width = 8
+
+styles = {"tab": {"height": "calc(98vh - 115px)"}}
+
+theme = {
+    "dark": True,
+    "detail": "#007439",
+    "primary": "#00EA64",
+    "secondary": "#6E6E6E",
+}
+
+##NGL viewer
+viewer = html.Div(
+    [dashbio.DashNgl(id="viewport", data=data_dict)],
+    style={
+        "display": "inline-block",
+        "width": "calc(100% - 500px)",
+        "float": "left",
+        "marginTop": "50px",
+        "marginRight": "50px",
+    },
+)
+
+def header_colors():
+    return {
+        'bg_color': '#e7625f',
+        'font_color': 'white'
+    }
+
+
+def description():
+    return 'Molecule visualization in 3D - perfect for viewing ' \
+           'biomolecules such as proteins, DNA and RNA. Includes ' \
+           'stick, cartoon, and sphere representations.'
+
+def layout():
+    return html.Div(
+        [
+            # header
+            html.Div(
+                children=[html.H1("NGL Protein Structure Viewer")],
+                style={"backgroundColor": "#3aaab2", "height": "7vh"},
+            ),
+            viewer,
+            # if you use dcc.Loading viewer the app goes into a mount loop!
```

Comment:
Could you please elaborate on this?

Reply:
Interesting this bug magically vanished at least on my laptop (Brave)*
I am going to add `dcc.Loading` in another commit.

*It might still not work on my office machine (Chromium)
going to try that on Monday

Reply:
@shammamah unfortunately the bug did not vanish completely the mount loop which leads to crashing the app is gone. However,  I realized that my feature: enabling reload from browser when molecule was selected before is not working since the component gets remounted after every new selection.

If you want to test on your own I updated my dash_ngl component
just uncomment dcc.Loading and you will see the component did mount
console.logs appearing after each new selection :-/